wgpu-native/src/device.rs

```diff
@@ -1215,7 +1215,7 @@ pub fn device_create_bind_group<B: GfxBackend>(
                         .buffers
                         .use_extend(&*buffer_guard, bb.buffer, (), usage)
                         .unwrap();
-                    assert!(buffer.usage.contains(usage));
+                    assert!(buffer.usage.contains(usage), format!("Invalid usage for buffer {:?}, {:?} is required", buffer.usage, usage));
```

Comment:
first of all, `format!` isn't needed when you are in the `assert!` macro
Secondly, we don't really need to print the `buffer.usage` here. There is no other way to fix it than going to the code responsible for buffer creation and checking/fixing the usage flags, so it's not like we'd be providing any info that the user wouldn't see anyway. We only need to print the `usage` here.

Reply:
How does the user know which buffer caused this ? I didn’t. Having which usages were passed was very helpful. I always think it’s better to give as much possible information during a crash as you can, so why not pass it ?